streams/src/main/java/org/apache/kafka/streams/processor/internals/StreamThread.java

```diff
@@ -158,9 +163,14 @@ public StreamThread(TopologyBuilder builder,
         this.processId = processId;
         this.partitionGrouper = config.getConfiguredInstance(StreamsConfig.PARTITION_GROUPER_CLASS_CONFIG, PartitionGrouper.class);
         this.streamsMetadataState = streamsMetadataState;
-
+        if (config.getLong(StreamsConfig.CACHE_MAX_BYTES_BUFFERING_CONFIG) < 0) {
+            log.info("Negative cache size passed in thread [{}]. Disabling cache", threadName);
```

Comment:
- should "0" be valid cache size?
- is seems, even of value is negative, we still instantiate `this.cache = new MemoryLRUCacheBytes(cacheSizeBytes);`below -- so why log message that says disabled? (it might effectively not do any caching as there is no cache size, but I would not call this disabled as be still instantiate and use the cache).